### PR TITLE
layers: Fix GPL RenderPass/Layout check

### PR DESCRIPTION
**Pull request by spencer-lunarg**
This adds `VUID-VkGraphicsPipelineCreateInfo-flags-06643` and `VUID-VkGraphicsPipelineCreateInfo-layout-06602`

In the process I found that the object tracker currently has some issue, spec issues were raised for them

### CODE REVIEW

File: layers/vulkan/generated/object_tracker.cpp [hunk 1]
```diff
@@ -1063,10 +1063,6 @@ bool ObjectLifetimes::PreCallValidateCreateGraphicsPipelines(VkDevice device, Vk
                     }
                 }
             }
-            skip |= ValidateObject(pCreateInfos[index0].layout, kVulkanObjectTypePipelineLayout, true, kVUIDUndefined,
```
Comment (spencer-lunarg) [inline layers/vulkan/generated/object_tracker.cpp:1066]: these are covered by 

VUID-VkGraphicsPipelineCreateInfo-dynamicRendering-06576
VUID-VkGraphicsPipelineCreateInfo-renderPass-06603
VUID-VkGraphicsPipelineCreateInfo-flags-06643
VUID-VkGraphicsPipelineCreateInfo-layout-06602
VUID-VkGraphicsPipelineCreateInfo-None-07826
VUID-VkGraphicsPipelineCreateInfo-flags-06642

File: layers/vulkan/generated/object_tracker.cpp [hunk 2]
```diff
@@ -5670,15 +5666,9 @@ bool ObjectLifetimes::PreCallValidateAcquireFullScreenExclusiveModeEXT(VkDevice
     return skip;
 }
 
-bool ObjectLifetimes::PreCallValidateReleaseFullScreenExclusiveModeEXT(VkDevice device, VkSwapchainKHR swapchain,
-                                                                       const ErrorObject& error_obj) const {
-    bool skip = false;
-    // Checked by chassis: device: kVUIDUndefined
-    skip |= ValidateObject(swapchain, kVulkanObjectTypeSwapchainKHR, false, kVUIDUndefined, kVUIDUndefined,
```
Comment (spencer-lunarg) [inline layers/vulkan/generated/object_tracker.cpp:5677]: will be re-fixed when spec is updated
https://gitlab.khronos.org/vulkan/vulkan/-/issues/3774

File: layers/vulkan/generated/object_tracker.cpp [hunk 3]
```diff
@@ -6610,8 +6600,7 @@ bool ObjectLifetimes::PreCallValidateCmdBuildMicromapsEXT(VkCommandBuffer comman
     if (pInfos) {
         for (uint32_t index0 = 0; index0 < infoCount; ++index0) {
             [[maybe_unused]] const Location index0_loc = error_obj.location.dot(Field::pInfos, index0);
-            skip |= ValidateObject(pInfos[index0].dstMicromap, kVulkanObjectTypeMicromapEXT, true, kVUIDUndefined, kVUIDUndefined,
```
Comment (spencer-lunarg) [inline layers/vulkan/generated/object_tracker.cpp:6613]: https://gitlab.khronos.org/vulkan/vulkan/-/issues/3775

File: layers/core_checks/cc_pipeline_graphics.cpp [hunk 4]
```diff
@@ -471,6 +467,32 @@ static bool ComparePipelineFragmentShadingRateStateCreateInfo(VkPipelineFragment
...
+            if (!enabled_features.dynamicRendering) {
+                skip |=
+                    LogError("VUID-VkGraphicsPipelineCreateInfo-dynamicRendering-06576", device,
+                             create_info_loc.dot(Field::renderPass), "is NULL, but the dynamicRendering feature was not enabled");
+            }
+        } else if (!pipeline.RenderPassState()) {
+            const auto gpl_info = vku::FindStructInPNextChain<VkGraphicsPipelineLibraryCreateInfoEXT>(pipeline.PNext());
+            const uint32_t flags_count =
+                (!gpl_info) ? 0
+                            : GetBitSetCount(gpl_info->flags & (VK_GRAPHICS_PIPELINE_LIBRARY_PRE_RASTERIZATION_SHADERS_BIT_EXT |
```
Comment (artem-lunarg) [inline layers/core_checks/cc_pipeline_graphics.cpp:484]: It looks like should work without `GetBitSetCount`, test filtered flags against zero.

**Comment by artem-lunarg** [comment]
@spencer-lunarg maybe you mentioned this already but `VUID-VkGraphicsPipelineCreateInfo-renderPass-06603` text in the spec looks broken for me.

**Review by artem-lunarg — APPROVED** [review]
LGTM, one potential simplification with `GetBitSetCount`.

**Comment by spencer-lunarg** [comment]
> VUID-VkGraphicsPipelineCreateInfo-renderPass-06603 text in the spec looks broken for me.

https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/6444

**Pull request merged**